src/site/content/en/blog/learn-to-bundle/index.md

```diff
@@ -0,0 +1,225 @@
+  <figcaption class="w-figcaption">
+    A figure showing how web bundle bundle web resources.
+  </figcaption>
+</figure>
+
+HTTP resources in a web bundle are indexed by request URLs, and can optionally
+come with signatures that vouch for the resources. Signatures allow browsers to
+understand and verify where each resource came from, and treats each as coming
+from its true origin. This is similar to how Signed HTTP Exchanges are handled
+by browsers. ([Signed HTTP
+Exchanges](https://developers.google.com/web/updates/2018/11/signed-exchanges)
+is a feature to sign a single HTTP resource.)
+
+
+In this article I will walk through what a web bundle is and how you can use it
+today. Let's begin the bundle tour.
+
+
+## Explaining web bundles
+
+To be more precise than my earlier definition, a web bundle is a [CBOR
+file](https://cbor.io/) with a `.wbn` extension (by convention) which packages
+HTTP resources in to a binary format, and is served with the
+`application/webbundle` MIME type. You can read more about this in the
+[Top-level
+structure](https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html#top-level)
+section of the draft.
+
+Some of its unique features are:
+
+* Encapsulates multiple pages, enabling bundling of a complete website into a single file.
+* JavaScript is executable (unlike MHTML).
+* Uses [HTTP
+  Variants](https://tools.ietf.org/id/draft-ietf-httpbis-variants-00.html) to do
+  content negotiation. In other words, internationalization can be done by using
+  the Accept-Language header even the bundle is used offline.
+* Can be cryptographically signed by the publisher's origin which allows the
+  browser to load the bundle's contents in the context of its origin.
+* Loads nearly instantly when it's served locally by the device.
+
+These features open multiple scenarios. One common scenario is the ability to
+build a self-contained web app that's easy to share and usable without an
+internet connection.
+
+Let's say you are on an airplane from Tokyo to San Francisco with your friend.
+You need to kill time for the nine hour flight but you don't like the in-flight
+entertainment. Your friend is playing an interesting web game called
+[Proxx](https://proxx.app/), and tells you that she downloaded the game as a web
+bundle before boarding the plane. It works flawlessly offline. Before web
+bundles, the story would end there and you would either have to take turns
+playing the game on your friend's device, or find something else to pass the
+time. But with web bundles, here is what you could do.
+
+1. Ask your friend to share the `.wbn` file of the game. For example the file
+   could easily be shared peer to peer using a file sharing app. If
+2. Open the `.wbn` file in a browser that supports web bundles.
+3. Start playing the game on your own device and try to beat your friend's high
+   score.
+
+Here is a video that explains this scenario.
+
+<figure class="w-figure w-figure--fullbleed">
```

Comment:
You'll want to replace this figure with [the `{% YouTube %}` component](https://web.dev/handbook/markup-media/#video).

```
{% YouTube 'xAujz66la3Y' %}
```